airflow/jobs/base_job.py

```diff
@@ -199,8 +188,9 @@ def heartbeat(self):
                 convert_camel_to_snake(self.__class__.__name__) + '_heartbeat_failure', 1,
                 1)
             self.log.exception("%s heartbeat got an exception", self.__class__.__name__)
-            # We didn't manage to heartbeat, so make sure that the timestamp isn't updated
-            self.latest_heartbeat = previous_heartbeat
```

Comment:
We need to keep this though, see https://issues.apache.org/jira/browse/AIRFLOW-5102